### PR TITLE
feat(ui): add buttons to add a bond or bridge

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add buttons to the network tab to create a bond or bridge provided the correct selection is made.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View the network tab for a machine.
- Add an alias if there isn't one already.
- Hover the bond button follow the directions on the tooltip to create a bond.
- Do the same for the bridge button.
- Select different combinations of interfaces and see that you get different tooltips with hints on both buttons.

## Fixes

Fixes: #2308.

### CODE REVIEW

File: ui/src/app/store/machine/utils/networking.ts [hunk 1]
```diff
@@ -59,6 +59,27 @@ export const getLinkInterface = (
   return getLinkInterfaceById(machine, link?.id);
 };
 
+/**
+ * Get an interface by id.
+ * @param machine - The nic's machine.
+ * @param linkId - A link's ID.
+ * @return An interface.
+ */
+export const getInterfaceById = (
```
Comment (huwshimi) [inline ui/src/app/store/machine/utils/networking.ts:68]: This was extracted from `machine/selectors.ts`

**Review by huwshimi — COMMENTED** [review]
This won't be reviewable until https://github.com/canonical-web-and-design/maas-ui/pull/2316 lands.

Also, this is not as scary as it looks, most of it is one tests file.

**Comment by huwshimi** [comment]
This is unblocked now.

File: ui/src/app/machines/views/MachineDetails/MachineNetwork/NetworkActions/NetworkActions.tsx [hunk 2]
```diff
@@ -30,25 +73,95 @@ const NetworkActions = ({
...
+        [expanded?.content === ExpandedState.ADD_PHYSICAL],
+      ],
+      label: "Add interface",
+      state: ExpandedState.ADD_PHYSICAL,
+    },
+    {
+      disabled: [
+        [isAllNetworkingDisabled, networkDisabledMessage],
+        [selected.length === 0, "No interfaces are selected"],
+        [selected.length === 1, "A bond must include more than one interface"],
```
Comment (anthonydillon) [inline ui/src/app/machines/views/MachineDetails/MachineNetwork/NetworkActions/NetworkActions.tsx:96]: Nice, very readable

**Review by anthonydillon — APPROVED** [review]
Code and QA :+1:

**Pull request merged**